### PR TITLE
remove the white space after the doctype according to the property co…

### LINKED ISSUE #7162: HTML minification with Doctype tag
### What version of `astro` are you using?

2.5.1

### Are you using an SSR adapter? If so, which one?

Node

### What package manager are you using?

npm

### What operating system are you using?

macOS

### What browser are you using?

Firefox

### Describe the Bug

After ran thru Astro HTML minification in 2.5.x release, I found curiosity why isn't docttype tag being minified?

```
<!DOCTYPE html>
<html ...>
```

### Link to Minimal Reproducible Example

NA

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by JerryWu1234**
…mpressHTML

fix #7162

## Changes
remove the white space after the doctype according to the property compressHTML

### CODE REVIEW

File: .changeset/tasty-stingrays-smile.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'astro': minor
```
Comment (matthewp) [inline .changeset/tasty-stingrays-smile.md:2]: Let's make this be `patch` as well.

**Comment by Princesseuh** [comment]
@matthewp Are there any streaming concerns with this? I feel like surely we made it so there's a newline after the doctype for a reason.

**Comment by wulinsheng123** [comment]
> @matthewp Are there any streaming concerns with this? I feel like surely we made it so there's a newline after the doctype for a reason.

will it break something ?

**Review by ematipico — APPROVED** [review]

**Comment by natemoo-re** [comment]
> will it break something ?

I don't think so. Newlines typically mean the browser will flush the buffer and attempt to parse the latest chunk as HTML, so in this case it will process the doctype when the first chunk is flushed. Seems like fine behavior to me.

**Review by natemoo-re — APPROVED** [review]

File: packages/astro/src/core/render/core.ts [hunk 2]
```diff
@@ -108,9 +108,10 @@ export type RenderPage = {
 	renderContext: RenderContext;
 	env: Environment;
 	apiContext?: APIContext;
+	isCompressHTMl?: boolean
 };
 
-export async function renderPage({ mod, renderContext, env, apiContext }: RenderPage) {
+export async function renderPage({ mod, renderContext, env, apiContext, isCompressHTMl = false }: RenderPage) {
```
Comment (bluwy) [inline packages/astro/src/core/render/core.ts:114]:
```suggestion
export async function renderPage({ mod, renderContext, env, apiContext, isCompressHTML = false }: RenderPage) {
```

The name isn't spelt right here (ant it's consumers).

**Comment by Princesseuh** [comment]
@matthewp The rest of the function is still faulty, the suggestion only correct one instance

**Comment by JerryWu1234** [comment]
> @matthewp The rest of the function is still faulty, the suggestion only correct one instance

I noticed that and i fixed it

**Comment by bluwy** [comment]
Thanks!

**Pull request merged**